system/protocols/mumble/protocol.py

```diff
@@ -582,37 +611,75 @@ def handle_msg_userstate(self, message):
         if message.name and message.session not in self.users:
             # Note: I'm not sure if message.name should ever be empty and
             # not in self.users - rakiru
-            self.users[message.session] = User(self,
-                                               message.session,
-                                               message.name,
-                                               self.channels[
-                                                   message.channel_id],
-                                               message.mute,
-                                               message.deaf,
-                                               message.suppress,
-                                               message.self_mute,
-                                               message.self_deaf,
-                                               message.priority_speaker,
-                                               message.recording)
+            # Note: RE: above note - Mumble desktop client source suggests not.
+            user = User(self,
```

Comment:
Usually when I do that I tend to split it up into groups of associated attrs, but I'm not overly picky honestly. It's already a godawful constructor. xD